src/Azure.Deployments.Extensibility.Providers.Graph/GraphProvider.cs

```diff
@@ -0,0 +1,403 @@
+
+                    if (value.GetArrayLength() > 0)
+                    {
+                        var firstValue = value.EnumerateArray().First();
+                        id = firstValue.GetProperty(Id).ToString();
+                    }
+                }
+
+            }
+            else if (response.StatusCode != HttpStatusCode.NotFound || 
+                !(uri.StartsWith("users") || NavigationPropertyByType.Contains(resourceType)))
+            {
+                throw new GraphHttpException((int)response.StatusCode, content);
+            }
+
+            return id;
+        }
+
+        /// <summary>
+        ///     Generate uri to send GET request to
+        ///     
+        ///     TODO: Update logic when we know what format the name has
+        ///             https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1998150
+        /// </summary>
+        /// <returns> 
+        ///     if type2 has alternate key: "type1/name1/type2/type2AlternateKey"
+        ///     if type2 doesn't have alternate key: "type1/name1/type2?$filter=displayName eq 'type2DisplayName'"
+        ///         where 'type2AlternateKey' and 'type2DisplayName' are fetched from properties
+        /// </returns>
+        public static string GenerateGetUri(string resourceType, JsonElement properties)
+        {
+            var types = GetTypes(resourceType);
+            var uri = BuildUri(types, properties);
+            var lastType = types.Last();
+
+            // Special case for entity collection
+            if (NavigationPropertyByType.Contains(resourceType))
+            {
+                var lastName = properties.GetProperty(Name).ToString().Split('/').Last();
+                uri = $"{uri}?$filter={Id} eq '{lastName}'";
+            }
+            else
+            {
+                var alternateKey = AlternateKeyByType.ContainsKey(lastType) ? AlternateKeyByType[lastType] : DisplayName;
+                // filter by id for groups/members
+                if (properties.TryGetProperty(alternateKey, out var alternateValue))
+                {
+                    if (AlternateKeyByType.ContainsKey(lastType))
+                    {
+                        uri = $"{uri}/{alternateValue}";
+                    }
+                    else
+                    {
+                        uri = $"{uri}?$filter={DisplayName} eq '{alternateValue}'";
+                    }
+                }
+                else
+                {
+                    throw new ExtensibilityException(InternalError, JsonPointer.Empty, "Alternate key does not exist.");
+                }
+            }
+                     
+
+            return uri;
+        }
+
+        /// <summary>
+        ///     Generate uri to send PATCH request to
+        ///     
+        ///     TODO: Update logic when we know what format the name has
+        ///             https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1998150
```

Comment:
Again, let's remove references to resources that only Microsoft contributors can see.